packages/components/src/icon/stories/index.story.tsx

```diff
@@ -47,26 +47,70 @@ FillColor.args = {
 	...Default.args,
 };
 
+/**
+ * When `icon` is a function, it will be passed the `size` prop and any other additional props.
+ */
 export const WithAFunction = Template.bind( {} );
 WithAFunction.args = {
 	...Default.args,
-	icon: () => (
-		<SVG>
-			<Path d="M5 4v3h5.5v12h3V7H19V4z" />
-		</SVG>
+	icon: ( { size }: { size?: number } ) => (
+		<img
+			width={ size }
+			height={ size }
+			src="https://s.w.org/style/images/about/WordPress-logotype-wmark.png"
+			alt="WordPress"
+		/>
 	),
 };
+WithAFunction.parameters = {
+	docs: {
+		source: {
+			code: `
+<Icon
+  icon={ ( { size } ) => (
+    <img
```

Comment:
That's understandable, but on the contrary, those snippets at least look good in the story:

![Screenshot 2024-11-26 at 14 09 09](https://github.com/user-attachments/assets/10935b2a-486e-4702-a5e0-eb20a4596321)

while the code snipped for the default story looks like this:

![Screenshot 2024-11-26 at 14 09 22](https://github.com/user-attachments/assets/b6dabee4-bd9a-4968-bbfb-eb72f8ac7961)

I wish they were all properly formatted.

Reply:
> I wish they were all properly formatted.

Right 😅 The automatic code snippet engine leaves a lot to be desired. I feel like we might need to start investing more in manual code snippets, since some of them are quite inscrutable.

Reply:
I wonder if we can layer prettier on top of it 😉